File: demo/password-demo.js
// Simulates a password prompt — puppetty must NOT answer it, and should
// cancel after --prompt-timeout.
import readline from 'node:readline';

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
rl.question('Enter your password: ', (pw) => {
  console.log(`SECURITY FAILURE: something typed a password (${pw.length} chars)!`);
  rl.close();
  process.exit(1);
});

File: engine-rs/examples/password_demo.rs
//! Simulates a password prompt — puppetty must NOT answer it, and should
//! cancel after --prompt-timeout.

use std::io::{BufRead, Write};

fn main() {
    print!("Enter your password: ");
    std::io::stdout().flush().unwrap();
    let mut pw = String::new();
    if std::io::stdin().lock().read_line(&mut pw).unwrap_or(0) > 0 {
        let pw = pw.trim_end_matches(['\r', '\n']);
        println!(
            "SECURITY FAILURE: something typed a password ({} chars)!",
            pw.len()
        );
        std::process::exit(1);
    }
}
